provider/pkg/await/await_test.go

```diff
@@ -4,18 +4,338 @@ package await
+			require.False(t, metadata.SkipAwaitLogic(cac.currentInputs), "Await logic should not execute when SkipWait is set")
+
+			// get the live object from the fake API Server
+			gvr, err := clients.GVRForGVK(cac.clientSet.RESTMapper, cac.currentInputs.GroupVersionKind())
+			require.NoError(t, err)
+			live, err := ctx.client.Tracker().Get(gvr, cac.currentOutputs.GetNamespace(), cac.currentOutputs.GetName())
+			require.NoError(t, err, "Live object should exist in the API Server")
+			pod, ok := live.(*unstructured.Unstructured)
+			require.True(t, ok)
+
+			// mutate the live object to simulate a observable status update.
+			err = unstructured.SetNestedField(pod.Object, "Running", "status", "phase")
+			require.NoError(t, err)
+			err = ctx.client.Tracker().Update(gvr, live, cac.currentOutputs.GetNamespace())
+			require.NoError(t, err)
+			return nil
+		}
+	}
+
+	awaitError := func(t *testing.T, ctx testCtx) createAwaiter {
+		return func(cac createAwaitConfig) error {
+			return serviceUnavailableErr
+		}
+	}
+
+	awaitUnexpected := func(t *testing.T, ctx testCtx) createAwaiter {
+		return func(cac createAwaitConfig) error {
+			require.Fail(t, "Unexpected call to awaiter")
+			return nil
+		}
+	}
+
+	// expectations
+
+	failed := func(target error) expectF {
+		return func(t *testing.T, ctx testCtx, actual *unstructured.Unstructured, err error) {
+			require.ErrorAs(t, err, &target)
+		}
+	}
+	previewed := func(ns, name string) expectF {
+		return func(t *testing.T, ctx testCtx, actual *unstructured.Unstructured, err error) {
+			require.NoError(t, err)
+			require.NotNil(t, actual)
+			require.Equal(t, ns, actual.GetNamespace(), "Object should have the expected namespace")
+			require.Equal(t, name, actual.GetName(), "Object should have the expected name")
+		}
+	}
+	created := func(ns, name string) expectF {
+		return func(t *testing.T, ctx testCtx, actual *unstructured.Unstructured, err error) {
+			require.NoError(t, err)
+			require.NotNil(t, actual)
+			require.Equal(t, ns, actual.GetNamespace(), "Object should have the expected namespace")
+			require.Equal(t, name, actual.GetName(), "Object should have the expected name")
+			gvr, err := clients.GVRForGVK(ctx.mapper, ctx.config.Inputs.GroupVersionKind())
+			require.NoError(t, err)
+			_, err = ctx.client.Tracker().Get(gvr, ns, name)
+			require.NoError(t, err, "Live object should exist in the API Server")
+		}
+	}
+	touched := func() expectF {
+		return func(t *testing.T, ctx testCtx, actual *unstructured.Unstructured, err error) {
+			require.NoError(t, err)
+			actualPhase, ok, err := unstructured.NestedString(actual.Object, "status", "phase")
+			require.NoError(t, err)
+			require.True(t, ok, "Object should have a status.phase field")
+			require.Equal(t, actualPhase, "Running", "Object should have status.phase of 'Running'")
+		}
+	}
+	logged := func() expectF {
+		return func(t *testing.T, ctx testCtx, actual *unstructured.Unstructured, err error) {
+			// TODO
```

Comment:
Can we expand on what the required work here is for the TODO and maybe also create a GH issue to track this?

Reply:
I have some follow-up tickets to test the other await methods. I will be sure to implement this at that time.